### ISSUE TITLE
Share Sheet: Use the Block Editor for content

### ISSUE THREAD

**Issue by twstokes** [wordpress-mobile/WordPress-iOS#20544]
### Expected behavior
Pages / Posts created from the Share Sheet use the Block Editor.

### Actual behavior
Pages / Posts created from the Share Sheet use the Classic Editor.

### Steps to reproduce the behavior

1. Create a page / post from the Share Sheet (see steps below)
2. Edit the page / post from the app

| Safari | Share Sheet | Classic Editor |
| - | - | - |
| ![image](https://user-images.githubusercontent.com/2092798/231908822-f8ba5492-e3c8-4305-b458-e3f8d7766ec2.png) | ![image](https://user-images.githubusercontent.com/2092798/231906177-cb76abde-ffd1-47e7-9a47-6147af204f88.png) | ![Simulator Screenshot - iPhone 14 Pro - 2023-04-13 at 19 35 16](https://user-images.githubusercontent.com/2092798/231906375-6af019bb-286c-4dd9-9f28-16bb309ea63f.png) |

| Safari | Share Sheet | Save as Draft |
| - | - | - |
| ![image](https://user-images.githubusercontent.com/2092798/231908837-0b44199d-df74-4edd-9c8f-5c1564b968c0.png) | ![image](https://user-images.githubusercontent.com/2092798/231907093-0d64e22a-c923-4839-be36-8e3c67c32401.png) | ![image](https://user-images.githubusercontent.com/2092798/231906831-3fca769e-2057-44cb-911f-7ae58f028995.png) |

**Comment by guarani** [comment]
+1 for this. Today I wanted to share multiple photos from the Photos app to the Jetpack app. Tapping the JP app in the share sheet does nothing (it does a slight animation as if to open the JP app and then stops). Only if I choose a single photo to share does it work by opening the classic editor.

With the change to the block editor, I think we would be able to add support for sharing multiple images into the app. Let me know though if it's a separate problem and in that case I'll create a separate issue for it 🙇

**Comment by twstokes** [comment]
> Let me know though if it's a separate problem and in that case I'll create a separate issue for it 🙇

👋 Hey @guarani! Good observation - IMO a separate issue with the `Share Extension` label would be helpful. I don't think it's entirely decided yet how this flow will be improved. ref: pc5zdG-11e-p2

**Comment by guarani** [comment]
Good idea @twstokes. I created an issue here: https://github.com/wordpress-mobile/WordPress-iOS/issues/21334 🙇

**Comment by guarani** [comment]
> Tapping the JP app in the share sheet does nothing (it does a slight animation as if to open the JP app and then stops)

I couldn't reproduce this anymore so https://github.com/wordpress-mobile/WordPress-iOS/issues/21334 is now closed.

**Comment by dangermattic** [comment]
Thanks for reporting! 👍
